modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/insertbutton/plugin.js

```diff
@@ -267,69 +198,97 @@
 				left: `${sideOffset}px`,
 				top: `${
 					selectionClientRect.y +
+					document.defaultView.pageYOffset +
 					(selectionClientRect.height - buttonHeight) / 2
 				}px`,
 			});
 		},
 
-		_tableData: {
-			columns: 3,
-			rows: 3,
-		},
+		init(editor) {
+			const button = CKEDITOR.dom.element.createFromHtml(template.source);
 
-		_tableToolbar: null,
+			const tableToolbar = this._createTableToolbar({editor});
 
-		_toolbarVisible: false,
+			const toolbar = this._createToolbar({editor, tableToolbar});
 
-		hide() {
-			if (this._button.getStyle('display') !== 'none') {
-				this._button.setStyle('display', 'none');
-			}
-			if (this._toolbar) {
-				this._toolbar.hide();
-			}
-			if (this._tableToolbar) {
-				this._tableToolbar.destroy();
-			}
-		},
+			const hide = () => {
+				button.addClass('hide');
 
-		init(editor) {
-			this.editor = editor;
-
-			this._eventListeners.push(
-				this.editor.on(
-					'afterCommandExec',
-					this._onAfterCommandExec.bind(this)
-				),
-				this.editor.on('blur', this._onBlur.bind(this)),
-				this.editor.on('change', this._onChange.bind(this)),
-				this.editor.on('contentDom', this._onContentDom.bind(this)),
-				this.editor.on('destroy', this._onDestroy.bind(this)),
-				this.editor.on('paste', this._onChange.bind(this)),
-				this.editor.on(
-					'selectionChange',
-					this._onSelectionChange.bind(this)
-				)
-			);
+				tableToolbar.hide();
+				toolbar.hide();
+			};
 
-			this._button = CKEDITOR.dom.element.createFromHtml(template.source);
+			const onFocusLoss = () => {
+				setTimeout(() => {
+					if (this._focusedEditorName !== editor.name) {
+						hide();
+					}
+				});
+			};
 
-			this._button.setStyles({
-				display: 'none',
-				position: 'absolute',
-			});
+			const eventListeners = [];
+
+			eventListeners.push(
+				button.on('blur', onFocusLoss),
+				editor.on('blur', onFocusLoss),
+				tableToolbar._view.parts.panel.on('focusout', onFocusLoss),
+				toolbar._view.parts.panel.on('focusout', onFocusLoss),
+
+				button.on('click', (event) => {
+					event.cancel();
+
+					tableToolbar.hide();
 
-			this._eventListeners.push(
-				this._button.on('click', this._onButtonClick.bind(this))
+					toolbar.attach(editor.getSelection());
+				}),
+
+				editor.on('afterCommandExec', hide),
+
+				editor.on('focus', () => {
+					this._focusedEditorName = editor.name;
+				}),
+
+				editor.on('change', hide),
+
+				editor.on('contentDom', () => {
+					const body = editor.document.getBody();
+
+					if (!body.contains(button)) {
+						body.append(button);
+					}
+				}),
+
+				editor.on('destroy', () => {
+					eventListeners.forEach((listener) => {
```

Comment:
This would be important in a case where there is code resolving with significant timeout, so it could be left dangling in case of SPA navigate or similar. But if you are referring to timeout hack from https://github.com/liferay-frontend/liferay-portal/pull/1575#discussion_r729250373, it resolves immediately so no cleanup is needed.